### PR TITLE
Earned Achievement Color Splash

### PR DESCRIPTION
**Pull request by seiyria**
Uses the achievement icon primary color for the earned award animation. 
This will extract the header where possible, or default to the red `#F12049`

### CODE REVIEW

File: src/shared/modules/badgeup-client/components/achievement-earned-popup/achievement-earned-popup.ts [hunk 1]
```diff
@@ -17,4 +25,34 @@ export class AchievementEarnedPopupComponent {
         this.achievement = this.navParams.get('achievement');
         this.earnedAwards = this.navParams.get('earnedAwards');
     }
+
+    public async ngOnInit() {
+
+        const url = this.achievement.meta.icon || BASE_ICON;
```
Comment (MarkHerhold) [inline src/shared/modules/badgeup-client/components/achievement-earned-popup/achievement-earned-popup.ts]: seems like we should be using the icon getter here

File: src/shared/modules/badgeup-client/components/achievement-earned-popup/achievement-earned-popup.ts [hunk 2]
```diff
@@ -1,14 +1,22 @@
...
-export class AchievementEarnedPopupComponent {
+export class AchievementEarnedPopupComponent implements OnInit {
     public achievement: Achievement;
     public earnedAwards: Award[];
+    public get icon() {
+        return this.achievement.meta.icon || BASE_ICON;
+    }
+
+    public baseColor = '#F12049';
+    public imageBase64: string;
```
Comment (MarkHerhold) [inline src/shared/modules/badgeup-client/components/achievement-earned-popup/achievement-earned-popup.ts]: unused variable?

File: src/shared/modules/badgeup-client/components/achievement-earned-popup/achievement-earned-popup.ts [hunk 3]
```diff
@@ -17,4 +25,34 @@ export class AchievementEarnedPopupComponent {
...
+        const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(this.baseColor);
+        const rgb = result ? {
+            r: parseInt(result[1], 16),
+            g: parseInt(result[2], 16),
+            b: parseInt(result[3], 16)
+        } : null;
+
+        const boxShadow = '0px 23px 18px';
+        if(!rgb) {
+            return `${boxShadow} rgba(241, 32, 73, 0.29)`;
```
Comment (MarkHerhold) [inline src/shared/modules/badgeup-client/components/achievement-earned-popup/achievement-earned-popup.ts]: can you add a comment or constant describing what `241, 32, 73, 0.29` is?

File: src/shared/modules/badgeup-client/core/badgeUpClient.ts [hunk 4]
```diff
@@ -119,7 +119,7 @@ export class BadgeUpClient {
                 er.data = e.data;
             }
         }
-
+
```
Comment (MarkHerhold) [inline src/shared/modules/badgeup-client/core/badgeUpClient.ts]: unneeded change

File: src/shared/modules/badgeup-client/components/achievement-earned-popup/achievement-earned-popup.scss [hunk 5]
```diff
@@ -114,7 +113,7 @@ achievement-earned-popup {
             & > :first-child {
                 animation-delay: 1.15s;
             }
-
+I
```
Comment (MarkHerhold) [inline src/shared/modules/badgeup-client/components/achievement-earned-popup/achievement-earned-popup.scss]: ?

File: src/shared/modules/badgeup-client/components/achievement-earned-popup/achievement-earned-popup.ts (hunk 1 above)
Comment (seiyria) [reply src/shared/modules/badgeup-client/components/achievement-earned-popup/achievement-earned-popup.ts]: yes! sorry, I was doing multiple things and forgot 😅

File: src/shared/modules/badgeup-client/components/achievement-earned-popup/achievement-earned-popup.ts (hunk 2 above)
Comment (seiyria) [reply src/shared/modules/badgeup-client/components/achievement-earned-popup/achievement-earned-popup.ts]: yeah! originally I was going to parse out the image and then display it, but I think it makes more sense to show it straight away and then get the color secondarily. thanks!

File: src/shared/modules/badgeup-client/components/achievement-earned-popup/achievement-earned-popup.ts (hunk 3 above)
Comment (seiyria) [reply src/shared/modules/badgeup-client/components/achievement-earned-popup/achievement-earned-popup.ts]: can do!

File: src/shared/modules/badgeup-client/components/achievement-earned-popup/achievement-earned-popup.scss (hunk 5 above)
Comment (seiyria) [reply src/shared/modules/badgeup-client/components/achievement-earned-popup/achievement-earned-popup.scss]: typo, I think!

**Pull request merged**